Sources/ParseSwift/Authentication/3rd Party/ParseFacebook.swift

```diff
@@ -0,0 +1,395 @@
+//
+//  ParseFacebook.swift
+//  ParseSwift
+//
+//  Created by Abdulaziz Alhomaidhi on 3/18/21.
+//  Copyright © 2021 Parse Community. All rights reserved.
+
+import Foundation
+#if canImport(Combine)
+import Combine
+#endif
+
+// swiftlint:disable line_length
+
+/**
+ Provides utility functions for working with Facebook User Authentication and `ParseUser`'s.
+ Be sure your Parse Server is configured for [sign in with Facebook](https://docs.parseplatform.org/parse-server/guide/#configuring-parse-server-for-sign-in-with-facebook).
+ For information on acquiring Facebook sign-in credentials to use with `ParseFacebook`, refer to [Facebook's Documentation](https://developers.facebook.com/docs/facebook-login/limited-login.
+ */
+public struct ParseFacebook<AuthenticatedUser: ParseUser>: ParseAuthentication {
+
+    /// Authentication keys required for Facebook authentication.
+    enum AuthenticationKeys: String, Codable {
+        case id // swiftlint:disable:this identifier_name
+        case authenticationToken
+        case accessToken
+        case expirationDate
+        
+        enum CodingKeys: String, CodingKey {
+          case id // swiftlint:disable:this identifier_name
+          case token
+          case accessToken = "access_token"
+          case expirationDate = "expiration_date"
+        }
```

Comment:
Can you add a space after this line

Reply:
can you please specify after which line I need to add the space? Also, do you want me to add a new line or just white space?

Reply:
It was a newline that was needed. When you add functions, you should separate them by 1 newline